examples/js/rtc/WebRTCRecorder.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * @author hypnosnova / https://github.com/HypnosNova
+ * @param canvas: in most time, it is renderer.domElement.
+ * @param options: {
+ *     fps: default is 60,
+ *     format: default is 'webm',
+ *     codecs: default is 'vp8'
+ * }
+ */
+
+THREE.WebRTCRecorder = function ( canvas, options ) {
+
+	if ( ! MediaSource ) {
+
+		console.error( "Your browser doesn't support WebRTC MediaSource. Try to use chrome or firefox browser." );
+		return;
+
+	}
+
+	if ( ! canvas.captureStream ) {
+
+		console.error( "Your browser doesn't support canvas captureStream. Try to use chrome or firefox browser." );
+		return;
+
+	}
+
+	this.video = document.createElement( "video" );
```

Comment:
Does a video element need to be built into the recorder? I think I'd prefer a `getUrl` function that returns a blob url to the recorded video (if the video has been stopped). As it is a data URL is generated and set on the video element every time the video is stopped with no way to revoke the generated URL.

To me the utility of this class is to handle the recording and conversion of a canvas to a video. Providing a video element or download function seems more the responsibility of the application that would use the `WebRTCRecorder`. But overall I like this addition!

Reply:
I update the recorder and remove video element. I also think it's much better now.